sdk-exporter/trace/shared/src/main/scala/org/typelevel/otel4s/sdk/exporter/otlp/OtlpHttpSpanExporter.scala

```diff
@@ -0,0 +1,221 @@
+
+      case Encoding.Protobuf =>
+        val content = Header.Raw(ci"Content-Type", "application/x-protobuf")
+        EntityEncoder.simple(content) { spans =>
+          val request: ExportTraceServiceRequest = ProtoCodecs.toProto(spans)
+          Chunk.array(request.toByteArray)
+        }
+    }
+
+  def exportSpans[G[_]: Foldable](spans: G[SpanData]): F[Unit] = {
+    val request =
+      Request[F](Method.POST, config.endpoint, HttpVersion.`HTTP/1.1`)
+        .withEntity(spans.toList)
+        .putHeaders(config.headers)
+
+    client
+      .run(request)
+      .use(response => logBody(response).unlessA(response.status.isSuccess))
+      .timeout(config.timeout)
+      .handleErrorWith { e =>
+        Console[F].errorln(
+          s"OtlpHttpSpanExporter: cannot export spans: ${e.getMessage}\n${e.getStackTrace.mkString("\n")}\n"
+        )
+      }
+  }
+
+  def flush: F[Unit] = Temporal[F].unit
+
+  private def logBody(response: Response[F]): F[Unit] =
+    for {
+      body <- response.bodyText.compile.string
+      _ <- Console[F].println(
+        s"[OtlpHttpSpanExporter/${config.encoding}] the request failed with [${response.status}]. Body: $body"
+      )
+    } yield ()
+
+}
+
+object OtlpHttpSpanExporter {
+
+  private object Defaults {
+    val Endpoint: Uri = uri"http://localhost:4318/v1/traces"
+    val Timeout: FiniteDuration = 10.seconds
+    val GzipCompression: Boolean = false
+  }
+
+  private final case class Config(
+      encoding: Encoding,
+      endpoint: Uri,
+      timeout: FiniteDuration,
+      headers: Headers,
+      gzipCompression: Boolean
+  )
+
+  sealed trait Encoding
+  object Encoding {
+    case object Json extends Encoding
+    case object Protobuf extends Encoding
+  }
+
+  /** A builder of [[OtlpHttpSpanExporter]] */
+  sealed trait Builder[F[_]] {
+
+    /** Sets the OTLP endpoint to connect to.
+      *
+      * The endpoint must start with either `http://` or `https://`, and include
+      * the full HTTP path.
+      *
+      * Default value is `http://localhost:4318/v1/traces`.
+      */
+    def withEndpoint(endpoint: Uri): Builder[F]
+
+    /** Sets the maximum time to wait for the collector to process an exported
+      * batch of spans.
+      *
+      * Default value is `10 seconds`.
+      */
+    def withTimeout(timeout: FiniteDuration): Builder[F]
+
+    /** Enables Gzip compression.
+      *
+      * The compression is disabled by default.
+      */
+    def enableGzip: Builder[F]
+
+    /** Disables Gzip compression. */
+    def disableGzip: Builder[F]
+
+    /** Adds headers to requests. */
+    def addHeaders(headers: Headers): Builder[F]
+
+    /** Creates a [[OtlpHttpSpanExporter]] using the configuration of this
+      * builder.
+      */
+    def build: Resource[F, SpanExporter[F]]
+  }
+
+  /** Creates a [[Builder]] of [[OtlpHttpSpanExporter]] with the default
+    * configuration.
+    */
+  def builder[F[_]: Async: Network: Console](encoding: Encoding): Builder[F] =
```

Comment:
Should we use `protobuf` by default? 
```suggestion
  def builder[F[_]: Async: Network: Console]: Builder[F] =
```

And the builder can offer some utility methods:
```scala
builder[IO].withEncoding(encoding) // use the given encoding
builder[IO].protobuf // use protobuf
builder[IO].json // use json
```